lib/Doctrine/ORM/Mapping/EmbeddedClassMapping.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ArrayAccess;
+
+use function property_exists;
+
+/** @template-implements ArrayAccess<string, mixed> */
+final class EmbeddedClassMapping implements ArrayAccess
+{
+    use ArrayAccessImplementation;
+
+    public string|bool|null $columnPrefix = null;
```

Comment:
Where does `bool` come from? From a PHP function that can return `false` on error and wasn't fixed yet?

Reply:
I must admit I didn't look into it until now. Apparently, it's the way to specify there should not be any column prefix: https://github.com/doctrine/orm/pull/1238. If you specify `null`, I think the column will be prefixed with the name of the entity the embedded class is embedded into.

Reply:
Oh and it's legal in union types with php 8.1, so I can use it!